admin/pkg/authtoken/authtoken.go

```diff
@@ -131,13 +131,27 @@ func (t *Token) SecretHash() []byte {
 	return hashed[:]
 }
 
+// Prefix returns a safe, partial display string for the token, e.g., "rill_usr_abcdefghij".
+// This works even if t.Secret is empty (all zeroes), as long as t.ID is set.
+func (t *Token) Prefix() string {
+	payload := make([]byte, 40)
+	copy(payload[0:16], t.ID[:])
+
+	// The remaining 24 bytes are already zero-initialized (Go default for slices)
+	return fmt.Sprintf("%s_%s_%s", Prefix, t.Type, marshalBase62(payload)[:10])
+}
+
 // marshalBase62 marshals a byte slice to a string of [0-9A-Za-z] characters.
 func marshalBase62(val []byte) string {
 	var i big.Int
 	i.SetBytes(val)
 	return i.Text(62)
 }
 
+func UnmarshalBase62(s string) ([]byte, bool) {
+	return unmarshalBase62(s)
+}
```

Comment:
1. This breaks the style guide – exported functions come before internal functions + should have a docstring.
2. Base62 is an implementation detail, as is the token format in general. Any parsing/serialization of tokens should happen inside this package. So this package shouldn't really expose this kind of function.